### ISSUE TITLE
Suggestion for a simple "histogram" distribution

### ISSUE THREAD

**Issue by mikkokemppainen** [tensorflow/probability#704]
I would find it useful to be able to form a probability distribution from a set of samples, similarly to the Empirical distribution, but in a continuous fashion, like a histogram. Such a distribution would be better suited for MCMC inference than a pointlike Empirical distribution.

Does there already exist a canonical way to form such distributions? If not, would it be beneficial to have one?

**Comment by brianwa84** [comment]
Sounds a bit like MixtureSameFamily with underlying Uniform?

Brian Patton | Software Engineer | bjp@google.com

**Comment by mikkokemppainen** [comment]
I agree, a mixture of Uniforms could be used. However, there is some amount of work in computing the mixture probabilities and bin endpoints from samples (with tf.histogram_fixed_width, say), and my suggestion would be to have a distribution, like Empirical, that would accept samples directly. Moreover, a reasonable number of bins could be in tens or even in hundreds, which to me sounds a bit cumbersome to be handled by means of a mixture.
